mlflow/utils/autologging_utils.py

```diff
@@ -812,6 +813,8 @@ def create_managed_run():
             managed_run.info.run_id,
             autologging_integration,
         )
+        if run_id_env_var_exists:
+            try_mlflow_log(mlflow.set_tags, tags)
```

Comment:
Fix for this error:

https://github.com/mlflow/mlflow/pull/4102/checks?check_run_id=1897305736#step:7:1605

```
AssertionError: Autologging run with id bfae40e28de046c4b17f590730c3f458 failed to set autologging tag with expected value. Expected: 'pytorch', Actual: 'None'
```

The error above occurs when running `pytorch/MNIST/example1` using `mlflow run`. `mlflow run` creates a new run and store its `run_id` in the `MLFLOW_RUN_ID` environment variable before running an entrypoint script. If `MLFLOW_RUN_ID` exists, `mlflow.start_run` just resumes a run with the specified id and **doesn't set tags**.

Reply:
Should we modify `mlflow.start_run` instead?

Reply:
@harupy Thanks for catching this. In this case, I think I'd rather we split the `mlflow.start_run(tags=...)` into two separate API calls; i.e.:

```
mlflow.start_run()
mlflow.set_tags(...)
```

This way, we don't have to special case the environment variable and rely on behaviors internal to `mlflow.start_run()`. I'm hesitant to modify the behavior of `mlflow.start_run()` to set the tags on an existing run, since the goal of this API mode is to resume an existing run, rather than mutate its data.

Can we separate the API calls and update the docs in `mlflow.start_run()` to indicate that `tags` are ignored if an existing run is being resumed? https://github.com/mlflow/mlflow/blob/4ba469ea5d67ca327c23a6e57d1fc2d2256c6f1c/mlflow/tracking/fluent.py#L132

Reply:
@dbczumar Thanks for the feedback. I'll make the change :)